pyproject.toml

```diff
@@ -0,0 +1,157 @@
+space_packet_parser = "^5.0.1"
+spiceypy = ">=6.0.0"
+xarray = '>=2024.10.0'
+numpy = "<=3"
+sammi-cdf = "^1.0"
+scipy = "^1.13"
+
+# Optional dependencies
+numpydoc = {version="^1.5.0", optional=true}
+openpyxl = {version=">=3.0.7", optional=true}
+pandas = {version=">=2.0.0", optional=true}
+pre-commit = {version="^3.3.3", optional=true}
+pydata-sphinx-theme = {version="*", optional=true}
+pytest = {version=">=6.2.5", optional=true}
+pytest-cov = {version="^4.0.0", optional=true}
+pytest-xdist = {version="^3.2", optional=true}
+ruff = {version="==0.2.1", optional=true}
+sphinx = {version="*", optional=true}
+sphinxcontrib-openapi = {version="^0.8.3", optional=true}
+mypy = {version="1.10.1", optional=true}
+requests = {version = "^2.32.3", optional = true}
+healpy = {version = "^1.18.0", optional = true}
+netcdf4 = {version ="^1.7.2",  optional = true}
+
+[tool.poetry.extras]
+dev = ["pre-commit", "ruff", "mypy"]
+doc = ["numpydoc", "pydata-sphinx-theme", "sphinx", "sphinxcontrib-openapi"]
+test = ["openpyxl", "pytest", "pytest-cov", "pytest-xdist", "requests", "netcdf4"]
+tools = ["openpyxl", "pandas"]
+map_visualization = ["healpy"]
+
+[project.urls]
+homepage = "https://github.com/IMAP-Science-Operations-Center"
+repository = "https://github.com/IMAP-Science-Operations-Center/imap_processing"
+
+[tool.pytest.ini_options]
+testpaths = [
+  "imap_processing/tests",
+]
+addopts = "-ra"
+filterwarnings = [
+    "ignore:Converting non-nanosecond:UserWarning:cdflib",
+    "ignore:datetime.datetime.utcfromtimestamp:DeprecationWarning:cdflib",
+    # Ignore Data_version warnings during tests because we default to v001
+    "ignore:No Data_version attribute found:UserWarning:imap_processing",
+]
+markers = [
+    "external_kernel: marks tests as requiring external SPICE kernels (deselect with '-m \"not external_kernel\"')",
+    "external_test_data: marks tests as requiring external test data (deselect with '-m \"not external_test_data\"')",
+    "use_test_metakernel: Mark test to use a test metakernel"
+]
+
+
+[tool.ruff]
+target-version = "py39"
+lint.select = ["B", "D", "E", "F", "I", "N", "S", "W", "PL", "PT", "UP", "RUF"]
+lint.ignore = [
+  "D104",  # Missing docstring in public package
+  "PLR0913",  # too-many-args error
+  "PLR0915",  # too-many-statements error
+  "PLR2004",  # Magic value in comparison
+  "RUF002",  # `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS) error
+  "RUF200",  # pyproject missing field (poetry doesn't follow the spec)
+  "S311",  # suspicious-non-cryptographic-random-usage
+]
+
+[tool.ruff.lint.per-file-ignores]
+# S603 unchecked input in subprocess call is fine in our tests
+# PT006 Wrong type passed to first argument of @pytest.mark.parametrize
+"*/tests/*" = ["D", "S101", "S603", "PT006"]
+
+[tool.ruff.lint.pydocstyle]
+convention = "numpy"
+
+[tool.poetry.scripts]
+imap_cli = 'imap_processing.cli:main'
+imap_xtce = 'imap_processing.ccsds.excel_to_xtce:main'
+
+[tool.codespell]
+ignore-words-list = "livetime"
```

Comment:
```suggestion
```

Until we encounter a purposely-misspelled word that is commonly used in this repo, we don't need this yet.